patchsets/ruby/1.8.7/default

```diff
@@ -1,3 +1,5 @@
 stdout-rouge-fix
 no_sslv2
 ssl_no_ec2m
+opensslv11x
+opensslv30
```

Comment:
how does it know to use the 1.1 patch or the 3.0 patch ?

Reply:
if you want it to suport OpenSSL 3.0, you need both patches (while on Ruby 1.8.7)
* Still compiles with my opensslv30 patch, even on openssl 1.1

My original patchset was just to support OpenSSL 1.1,  but later wanted to get Ruby 2.5 working on OpenSSL 3.0 (so came up with that patch).  I thought it might work on top of the 11x patch work, and it did.

I have not tested how well this works in a OpenSSL 1.0.x environment though.

Reply:
Ruby 2.4 was the first version to support OpenSSL 1.1.x, (most of the code I backported came from there or newer),  so 2.4 or newer would be opensslv3.0 only (I have only tested on the versions that I updated the patchsets on).